op-batcher/batcher/driver.go

```diff
@@ -476,9 +475,15 @@ func (l *BatchSubmitter) mainLoop(ctx context.Context, receiptsCh chan txmgr.TxR
 			l.publishStateToL1(queue, receiptsCh, daGroup, l.Config.PollInterval)
 
 		case <-ctx.Done():
+			queueCancel()
 			if err := queue.Wait(); err != nil {
-				l.Log.Error("error waiting for transactions to complete", "err", err)
+				if !errors.Is(err, context.Canceled) {
+					l.Log.Error("error waiting for transactions to complete", "err", err)
+				}
 			}
+			throttlingLoopCancel()
+			receiptsLoopCancel()
+			l.wg.Done()
```

Comment:
what's the difference to calling them in a `defer` at the top?

Reply:
No difference -- there's only one return statement in this function. I wanted to put all of the cleanup work in one spot to make it a bit easier to reason about.

Reply:
also works, just wanted to make sure I don't misunderstand